INFO:sagemaker:Creating training-job with name: tf2-object-detection-2023-03-21-05-42-14-527
2023-03-21 05:42:28 Starting - Starting the training job...
2023-03-21 05:42:54 Starting - Preparing the instances for training......
2023-03-21 05:43:51 Downloading - Downloading input data...
2023-03-21 05:44:11 Training - Downloading the training image...............
2023-03-21 05:46:52 Training - Training image download completed. Training in progress...2023-03-21 05:47:20,804 sagemaker-training-toolkit INFO     No Neurons detected (normal if no neurons installed)
2023-03-21 05:47:20,838 sagemaker-training-toolkit INFO     No Neurons detected (normal if no neurons installed)
2023-03-21 05:47:20,871 sagemaker-training-toolkit INFO     No Neurons detected (normal if no neurons installed)
2023-03-21 05:47:20,885 sagemaker-training-toolkit INFO     Invoking user script
Training Env:
{
    "additional_framework_parameters": {},
    "channel_input_dirs": {
        "train": "/opt/ml/input/data/train",
        "val": "/opt/ml/input/data/val"
    },
    "current_host": "algo-1",
    "current_instance_group": "homogeneousCluster",
    "current_instance_group_hosts": [
        "algo-1"
    ],
    "current_instance_type": "ml.p3.2xlarge",
    "distribution_hosts": [],
    "distribution_instance_groups": [],
    "framework_module": null,
    "hosts": [
        "algo-1"
    ],
    "hyperparameters": {
        "model_dir": "/opt/training",
        "num_train_steps": "2000",
        "pipeline_config_path": "pipeline.config",
        "sample_1_of_n_eval_examples": "1"
    },
    "input_config_dir": "/opt/ml/input/config",
    "input_data_config": {
        "train": {
            "TrainingInputMode": "File",
            "S3DistributionType": "FullyReplicated",
            "RecordWrapperType": "None"
        },
        "val": {
            "TrainingInputMode": "File",
            "S3DistributionType": "FullyReplicated",
            "RecordWrapperType": "None"
        }
    },
    "input_dir": "/opt/ml/input",
    "instance_groups": [
        "homogeneousCluster"
    ],
    "instance_groups_dict": {
        "homogeneousCluster": {
            "instance_group_name": "homogeneousCluster",
            "instance_type": "ml.p3.2xlarge",
            "hosts": [
                "algo-1"
            ]
        }
    },
    "is_hetero": false,
    "is_master": true,
    "is_modelparallel_enabled": null,
    "is_smddpmprun_installed": false,
    "job_name": "tf2-object-detection-2023-03-21-05-42-14-527",
    "log_level": 20,
    "master_hostname": "algo-1",
    "model_dir": "/opt/ml/model",
    "module_dir": "s3://sagemaker-us-east-1-472643955631/tf2-object-detection-2023-03-21-05-42-14-527/source/sourcedir.tar.gz",
    "module_name": "run_training.sh",
    "network_interface_name": "eth0",
    "num_cpus": 8,
    "num_gpus": 1,
    "num_neurons": 0,
    "output_data_dir": "/opt/ml/output/data",
    "output_dir": "/opt/ml/output",
    "output_intermediate_dir": "/opt/ml/output/intermediate",
    "resource_config": {
        "current_host": "algo-1",
        "current_instance_type": "ml.p3.2xlarge",
        "current_group_name": "homogeneousCluster",
        "hosts": [
            "algo-1"
        ],
        "instance_groups": [
            {
                "instance_group_name": "homogeneousCluster",
                "instance_type": "ml.p3.2xlarge",
                "hosts": [
                    "algo-1"
                ]
            }
        ],
        "network_interface_name": "eth0"
    },
    "user_entry_point": "run_training.sh"
}
Environment variables:
SM_HOSTS=["algo-1"]
SM_NETWORK_INTERFACE_NAME=eth0
SM_HPS={"model_dir":"/opt/training","num_train_steps":"2000","pipeline_config_path":"pipeline.config","sample_1_of_n_eval_examples":"1"}
SM_USER_ENTRY_POINT=run_training.sh
SM_FRAMEWORK_PARAMS={}
SM_RESOURCE_CONFIG={"current_group_name":"homogeneousCluster","current_host":"algo-1","current_instance_type":"ml.p3.2xlarge","hosts":["algo-1"],"instance_groups":[{"hosts":["algo-1"],"instance_group_name":"homogeneousCluster","instance_type":"ml.p3.2xlarge"}],"network_interface_name":"eth0"}
SM_INPUT_DATA_CONFIG={"train":{"RecordWrapperType":"None","S3DistributionType":"FullyReplicated","TrainingInputMode":"File"},"val":{"RecordWrapperType":"None","S3DistributionType":"FullyReplicated","TrainingInputMode":"File"}}
SM_OUTPUT_DATA_DIR=/opt/ml/output/data
SM_CHANNELS=["train","val"]
SM_CURRENT_HOST=algo-1
SM_CURRENT_INSTANCE_TYPE=ml.p3.2xlarge
SM_CURRENT_INSTANCE_GROUP=homogeneousCluster
SM_CURRENT_INSTANCE_GROUP_HOSTS=["algo-1"]
SM_INSTANCE_GROUPS=["homogeneousCluster"]
SM_INSTANCE_GROUPS_DICT={"homogeneousCluster":{"hosts":["algo-1"],"instance_group_name":"homogeneousCluster","instance_type":"ml.p3.2xlarge"}}
SM_DISTRIBUTION_INSTANCE_GROUPS=[]
SM_IS_HETERO=false
SM_MODULE_NAME=run_training.sh
SM_LOG_LEVEL=20
SM_FRAMEWORK_MODULE=
SM_INPUT_DIR=/opt/ml/input
SM_INPUT_CONFIG_DIR=/opt/ml/input/config
SM_OUTPUT_DIR=/opt/ml/output
SM_NUM_CPUS=8
SM_NUM_GPUS=1
SM_NUM_NEURONS=0
SM_MODEL_DIR=/opt/ml/model
SM_MODULE_DIR=s3://sagemaker-us-east-1-472643955631/tf2-object-detection-2023-03-21-05-42-14-527/source/sourcedir.tar.gz
SM_TRAINING_ENV={"additional_framework_parameters":{},"channel_input_dirs":{"train":"/opt/ml/input/data/train","val":"/opt/ml/input/data/val"},"current_host":"algo-1","current_instance_group":"homogeneousCluster","current_instance_group_hosts":["algo-1"],"current_instance_type":"ml.p3.2xlarge","distribution_hosts":[],"distribution_instance_groups":[],"framework_module":null,"hosts":["algo-1"],"hyperparameters":{"model_dir":"/opt/training","num_train_steps":"2000","pipeline_config_path":"pipeline.config","sample_1_of_n_eval_examples":"1"},"input_config_dir":"/opt/ml/input/config","input_data_config":{"train":{"RecordWrapperType":"None","S3DistributionType":"FullyReplicated","TrainingInputMode":"File"},"val":{"RecordWrapperType":"None","S3DistributionType":"FullyReplicated","TrainingInputMode":"File"}},"input_dir":"/opt/ml/input","instance_groups":["homogeneousCluster"],"instance_groups_dict":{"homogeneousCluster":{"hosts":["algo-1"],"instance_group_name":"homogeneousCluster","instance_type":"ml.p3.2xlarge"}},"is_hetero":false,"is_master":true,"is_modelparallel_enabled":null,"is_smddpmprun_installed":false,"job_name":"tf2-object-detection-2023-03-21-05-42-14-527","log_level":20,"master_hostname":"algo-1","model_dir":"/opt/ml/model","module_dir":"s3://sagemaker-us-east-1-472643955631/tf2-object-detection-2023-03-21-05-42-14-527/source/sourcedir.tar.gz","module_name":"run_training.sh","network_interface_name":"eth0","num_cpus":8,"num_gpus":1,"num_neurons":0,"output_data_dir":"/opt/ml/output/data","output_dir":"/opt/ml/output","output_intermediate_dir":"/opt/ml/output/intermediate","resource_config":{"current_group_name":"homogeneousCluster","current_host":"algo-1","current_instance_type":"ml.p3.2xlarge","hosts":["algo-1"],"instance_groups":[{"hosts":["algo-1"],"instance_group_name":"homogeneousCluster","instance_type":"ml.p3.2xlarge"}],"network_interface_name":"eth0"},"user_entry_point":"run_training.sh"}
SM_USER_ARGS=["--model_dir","/opt/training","--num_train_steps","2000","--pipeline_config_path","pipeline.config","--sample_1_of_n_eval_examples","1"]
SM_OUTPUT_INTERMEDIATE_DIR=/opt/ml/output/intermediate
SM_CHANNEL_TRAIN=/opt/ml/input/data/train
SM_CHANNEL_VAL=/opt/ml/input/data/val
SM_HP_MODEL_DIR=/opt/training
SM_HP_NUM_TRAIN_STEPS=2000
SM_HP_PIPELINE_CONFIG_PATH=pipeline.config
SM_HP_SAMPLE_1_OF_N_EVAL_EXAMPLES=1
PYTHONPATH=/opt/ml/code:/usr/local/bin:/usr/lib/python38.zip:/usr/lib/python3.8:/usr/lib/python3.8/lib-dynload:/usr/local/lib/python3.8/dist-packages:/usr/lib/python3/dist-packages
Invoking script with the following command:
/bin/sh -c "./run_training.sh --model_dir /opt/training --num_train_steps 2000 --pipeline_config_path pipeline.config --sample_1_of_n_eval_examples 1"
2023-03-21 05:47:20,886 sagemaker-training-toolkit INFO     Exceptions not imported for SageMaker Debugger as it is not installed.
===TRAINING THE MODEL==
INFO:tensorflow:Using MirroredStrategy with devices ('/job:localhost/replica:0/task:0/device:GPU:0',)
I0321 05:47:28.993580 139931598513984 mirrored_strategy.py:374] Using MirroredStrategy with devices ('/job:localhost/replica:0/task:0/device:GPU:0',)
INFO:tensorflow:Maybe overwriting train_steps: 2000
I0321 05:47:28.997253 139931598513984 config_util.py:552] Maybe overwriting train_steps: 2000
INFO:tensorflow:Maybe overwriting use_bfloat16: False
I0321 05:47:28.997441 139931598513984 config_util.py:552] Maybe overwriting use_bfloat16: False
WARNING:tensorflow:From /usr/local/lib/python3.8/dist-packages/object_detection/model_lib_v2.py:563: StrategyBase.experimental_distribute_datasets_from_function (from tensorflow.python.distribute.distribute_lib) is deprecated and will be removed in a future version.
Instructions for updating:
rename to distribute_datasets_from_function
W0321 05:47:29.050864 139931598513984 deprecation.py:350] From /usr/local/lib/python3.8/dist-packages/object_detection/model_lib_v2.py:563: StrategyBase.experimental_distribute_datasets_from_function (from tensorflow.python.distribute.distribute_lib) is deprecated and will be removed in a future version.
Instructions for updating:
rename to distribute_datasets_from_function
Traceback (most recent call last):
  File "model_main_tf2.py", line 114, in <module>
    tf.compat.v1.app.run()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/platform/app.py", line 36, in run
    _run(main=main, argv=argv, flags_parser=_parse_flags_tolerate_undef)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "model_main_tf2.py", line 105, in main
    model_lib_v2.train_loop(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/model_lib_v2.py", line 563, in train_loop
    train_input = strategy.experimental_distribute_datasets_from_function(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/util/deprecation.py", line 357, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 1198, in experimental_distribute_datasets_from_function
    return self.distribute_datasets_from_function(dataset_fn, options)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 1189, in distribute_datasets_from_function
    return self._extended._distribute_datasets_from_function(  # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/mirrored_strategy.py", line 613, in _distribute_datasets_from_function
    return input_util.get_distributed_datasets_from_function(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/input_util.py", line 132, in get_distributed_datasets_from_function
    return input_lib.DistributedDatasetsFromFunction(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/input_lib.py", line 1372, in __init__
    self.build()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/input_lib.py", line 1393, in build
    _create_datasets_from_function_with_input_context(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/input_lib.py", line 1875, in _create_datasets_from_function_with_input_context
    dataset = dataset_fn(ctx)
  File "/usr/local/lib/python3.8/dist-packages/object_detection/model_lib_v2.py", line 554, in train_dataset_fn
    train_input = inputs.train_input(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/inputs.py", line 908, in train_input
    dataset = INPUT_BUILDER_UTIL_MAP['dataset_build'](
  File "/usr/local/lib/python3.8/dist-packages/object_detection/builders/dataset_builder.py", line 209, in build
    decoder = decoder_builder.build(input_reader_config)
  File "/usr/local/lib/python3.8/dist-packages/object_detection/builders/decoder_builder.py", line 51, in build
    decoder = tf_example_decoder.TfExampleDecoder(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/data_decoders/tf_example_decoder.py", line 458, in __init__
    _ClassTensorHandler(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/data_decoders/tf_example_decoder.py", line 92, in __init__
    name_to_id = label_map_util.get_label_map_dict(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/utils/label_map_util.py", line 211, in get_label_map_dict
    label_map = load_labelmap(label_map_path_or_proto)
  File "/usr/local/lib/python3.8/dist-packages/object_detection/utils/label_map_util.py", line 172, in load_labelmap
    label_map_string = fid.read()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/lib/io/file_io.py", line 116, in read
    self._preread_check()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/lib/io/file_io.py", line 77, in _preread_check
    self._read_buf = _pywrap_file_io.BufferedInputStream(
tensorflow.python.framework.errors_impl.NotFoundError: PATH_TO_BE_CONFIGURED/label_map.txt; No such file or directory
==EVALUATING THE MODEL==
WARNING:tensorflow:Forced number of epochs for all eval validations to be 1.
W0321 05:47:33.870966 140516824635200 model_lib_v2.py:1089] Forced number of epochs for all eval validations to be 1.
INFO:tensorflow:Maybe overwriting sample_1_of_n_eval_examples: None
I0321 05:47:33.871331 140516824635200 config_util.py:552] Maybe overwriting sample_1_of_n_eval_examples: None
INFO:tensorflow:Maybe overwriting use_bfloat16: False
I0321 05:47:33.871438 140516824635200 config_util.py:552] Maybe overwriting use_bfloat16: False
INFO:tensorflow:Maybe overwriting eval_num_epochs: 1
I0321 05:47:33.871533 140516824635200 config_util.py:552] Maybe overwriting eval_num_epochs: 1
WARNING:tensorflow:Expected number of evaluation epochs is 1, but instead encountered `eval_on_train_input_config.num_epochs` = 0. Overwriting `num_epochs` to 1.
W0321 05:47:33.871696 140516824635200 model_lib_v2.py:1106] Expected number of evaluation epochs is 1, but instead encountered `eval_on_train_input_config.num_epochs` = 0. Overwriting `num_epochs` to 1.
Traceback (most recent call last):
  File "model_main_tf2.py", line 114, in <module>
    tf.compat.v1.app.run()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/platform/app.py", line 36, in run
    _run(main=main, argv=argv, flags_parser=_parse_flags_tolerate_undef)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "model_main_tf2.py", line 81, in main
    model_lib_v2.eval_continuously(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/model_lib_v2.py", line 1123, in eval_continuously
    inputs.eval_input(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/inputs.py", line 1071, in eval_input
    dataset = INPUT_BUILDER_UTIL_MAP['dataset_build'](
  File "/usr/local/lib/python3.8/dist-packages/object_detection/builders/dataset_builder.py", line 209, in build
    decoder = decoder_builder.build(input_reader_config)
  File "/usr/local/lib/python3.8/dist-packages/object_detection/builders/decoder_builder.py", line 51, in build
    decoder = tf_example_decoder.TfExampleDecoder(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/data_decoders/tf_example_decoder.py", line 458, in __init__
    _ClassTensorHandler(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/data_decoders/tf_example_decoder.py", line 92, in __init__
    name_to_id = label_map_util.get_label_map_dict(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/utils/label_map_util.py", line 211, in get_label_map_dict
    label_map = load_labelmap(label_map_path_or_proto)
  File "/usr/local/lib/python3.8/dist-packages/object_detection/utils/label_map_util.py", line 172, in load_labelmap
    label_map_string = fid.read()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/lib/io/file_io.py", line 116, in read
    self._preread_check()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/lib/io/file_io.py", line 77, in _preread_check
    self._read_buf = _pywrap_file_io.BufferedInputStream(
tensorflow.python.framework.errors_impl.NotFoundError: PATH_TO_BE_CONFIGURED/label_map.txt; No such file or directory
==EXPORTING THE MODEL==
WARNING:tensorflow:From /usr/local/lib/python3.8/dist-packages/tensorflow/python/autograph/pyct/static_analysis/liveness.py:83: Analyzer.lamba_check (from tensorflow.python.autograph.pyct.static_analysis.liveness) is deprecated and will be removed after 2023-09-23.
Instructions for updating:
Lambda fuctions will be no more assumed to be used in the statement where they are used, or at least in the same block. https://github.com/tensorflow/tensorflow/issues/56089
W0321 05:47:40.904543 140680654456640 deprecation.py:350] From /usr/local/lib/python3.8/dist-packages/tensorflow/python/autograph/pyct/static_analysis/liveness.py:83: Analyzer.lamba_check (from tensorflow.python.autograph.pyct.static_analysis.liveness) is deprecated and will be removed after 2023-09-23.
Instructions for updating:
Lambda fuctions will be no more assumed to be used in the statement where they are used, or at least in the same block. https://github.com/tensorflow/tensorflow/issues/56089
WARNING:tensorflow:From /usr/local/lib/python3.8/dist-packages/tensorflow/python/autograph/impl/api.py:458: calling map_fn_v2 (from tensorflow.python.ops.map_fn) with back_prop=False is deprecated and will be removed in a future version.
Instructions for updating:
back_prop=False is deprecated. Consider using tf.stop_gradient instead.
Instead of:
results = tf.map_fn(fn, elems, back_prop=False)
Use:
results = tf.nest.map_structure(tf.stop_gradient, tf.map_fn(fn, elems))
W0321 05:47:40.985601 140680654456640 deprecation.py:623] From /usr/local/lib/python3.8/dist-packages/tensorflow/python/autograph/impl/api.py:458: calling map_fn_v2 (from tensorflow.python.ops.map_fn) with back_prop=False is deprecated and will be removed in a future version.
Instructions for updating:
back_prop=False is deprecated. Consider using tf.stop_gradient instead.
Instead of:
results = tf.map_fn(fn, elems, back_prop=False)
Use:
results = tf.nest.map_structure(tf.stop_gradient, tf.map_fn(fn, elems))
INFO:tensorflow:depth of additional conv before box predictor: 0
I0321 05:47:47.813663 140680654456640 convolutional_keras_box_predictor.py:152] depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
I0321 05:47:47.813924 140680654456640 convolutional_keras_box_predictor.py:152] depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
I0321 05:47:47.814145 140680654456640 convolutional_keras_box_predictor.py:152] depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
I0321 05:47:47.814345 140680654456640 convolutional_keras_box_predictor.py:152] depth of additional conv before box predictor: 0
INFO:tensorflow:depth of additional conv before box predictor: 0
I0321 05:47:47.814538 140680654456640 convolutional_keras_box_predictor.py:152] depth of additional conv before box predictor: 0
WARNING:tensorflow:From /usr/local/lib/python3.8/dist-packages/tensorflow/python/util/dispatch.py:1176: batch_gather (from tensorflow.python.ops.array_ops) is deprecated and will be removed after 2017-10-25.
Instructions for updating:
`tf.batch_gather` is deprecated, please use `tf.gather` with `batch_dims=tf.rank(indices) - 1` instead.
W0321 05:48:09.897603 140680654456640 deprecation.py:350] From /usr/local/lib/python3.8/dist-packages/tensorflow/python/util/dispatch.py:1176: batch_gather (from tensorflow.python.ops.array_ops) is deprecated and will be removed after 2017-10-25.
Instructions for updating:
`tf.batch_gather` is deprecated, please use `tf.gather` with `batch_dims=tf.rank(indices) - 1` instead.
Traceback (most recent call last):
  File "exporter_main_v2.py", line 164, in <module>
    app.run(main)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "exporter_main_v2.py", line 157, in main
    exporter_lib_v2.export_inference_graph(
  File "/usr/local/lib/python3.8/dist-packages/object_detection/exporter_lib_v2.py", line 271, in export_inference_graph
    status.assert_existing_objects_matched()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/checkpoint/checkpoint.py", line 951, in assert_existing_objects_matched
    raise AssertionError(
AssertionError: No checkpoint specified (save_path=None); nothing is being restored.
mv: cannot stat '/tmp/exported/saved_model': No such file or directory
2023-03-21 05:48:29,120 sagemaker-training-toolkit ERROR    Reporting training FAILURE
2023-03-21 05:48:29,121 sagemaker-training-toolkit ERROR    ExecuteUserScriptError:
ExitCode 1
ErrorMessage ""
Command "/bin/sh -c ./run_training.sh --model_dir /opt/training --num_train_steps 2000 --pipeline_config_path pipeline.config --sample_1_of_n_eval_examples 1"
2023-03-21 05:48:29,121 sagemaker-training-toolkit ERROR    Encountered exit_code 1

2023-03-21 05:48:43 Uploading - Uploading generated training model
2023-03-21 05:48:43 Failed - Training job failed
---------------------------------------------------------------------------
UnexpectedStatusException                 Traceback (most recent call last)
Cell In[15], line 24
      1 tensorboard_output_config = sagemaker.debugger.TensorBoardOutputConfig(
      2     s3_output_path=tensorboard_s3_prefix,
      3     container_local_output_path='/opt/training/'
      4 )
      6 estimator = CustomFramework(
      7     role=role,
      8     image_uri=container,
   (...)
     21     base_job_name='tf2-object-detection'
     22 )
---> 24 estimator.fit(inputs)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/sagemaker/workflow/pipeline_context.py:272, in runnable_by_pipeline.<locals>.wrapper(*args, **kwargs)
    268         return context
    270     return _StepArguments(retrieve_caller_name(self_instance), run_func, *args, **kwargs)
--> 272 return run_func(*args, **kwargs)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/sagemaker/estimator.py:1156, in EstimatorBase.fit(self, inputs, wait, logs, job_name, experiment_config)
   1154 self.jobs.append(self.latest_training_job)
   1155 if wait:
-> 1156     self.latest_training_job.wait(logs=logs)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/sagemaker/estimator.py:2297, in _TrainingJob.wait(self, logs)
   2295 # If logs are requested, call logs_for_jobs.
   2296 if logs != "None":
-> 2297     self.sagemaker_session.logs_for_job(self.job_name, wait=True, log_type=logs)
   2298 else:
   2299     self.sagemaker_session.wait_for_job(self.job_name)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/sagemaker/session.py:4214, in Session.logs_for_job(self, job_name, wait, poll, log_type)
   4211             last_profiler_rule_statuses = profiler_rule_statuses
   4213 if wait:
-> 4214     self._check_job_status(job_name, description, "TrainingJobStatus")
   4215     if dot:
   4216         print()

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/sagemaker/session.py:3747, in Session._check_job_status(self, job, desc, status_key_name)
   3741 if "CapacityError" in str(reason):
   3742     raise exceptions.CapacityError(
   3743         message=message,
   3744         allowed_statuses=["Completed", "Stopped"],
   3745         actual_status=status,
   3746     )
-> 3747 raise exceptions.UnexpectedStatusException(
   3748     message=message,
   3749     allowed_statuses=["Completed", "Stopped"],
   3750     actual_status=status,
   3751 )

UnexpectedStatusException: Error for Training job tf2-object-detection-2023-03-21-05-42-14-527: Failed. Reason: AlgorithmError: ExecuteUserScriptError:
ExitCode 1
ErrorMessage ""
Command "/bin/sh -c ./run_training.sh --model_dir /opt/training --num_train_steps 2000 --pipeline_config_path pipeline.config --sample_1_of_n_eval_examples 1", exit code: 1